### ISSUE TITLE
Ditto does not acknowledge QoS 1 or 2 messages for MQTT 5 when there are several sources

### ISSUE THREAD

**Issue by dimabarbul** [eclipse-ditto/ditto#1768]
Ditto does not acknowledge QoS 1 or 2 messages for MQTT 5 when there are several sources.

Setup:

- running Ditto
- running MQTT 5 broker, e.g. mosquitto
- MQTT 5 connection that, for example, updates device twin, with following config:
    - static specificConfig.clientId
    - qos 2 in source

Steps:
- send MQTT 5 message that should lead to device twin updated, with QoS 1 or 2

    ```shell
    # data topic
    mosquitto_pub --id CLI -V mqttv5 -t data -m $RANDOM -q 1
    # ditto topic
    mosquitto_pub --id CLI -V mqttv5 -t ditto -m '{"topic":"test/thing-01/things/twin/commands/merge", "headers": {"content-type":"application/merge-patch+json"}, "path":"/features/ditto/properties/value", "value":'$RANDOM'}' -q 2
    ```

Relevant logs from mosquitto:

```text
1696746966: New connection from 127.0.0.1:36164 on port 1883.
1696746966: New client connected from 127.0.0.1:36164 as CLI (p5, c1, k60).
1696746966: No will message specified.
1696746966: Sending CONNACK to CLI (0, 0)
1696746966: Received PUBLISH from CLI (d0, q2, r0, m1, 'data', ... (5 bytes))
1696746966: Sending PUBREC to CLI (m1, rc0)
1696746966: Received PUBREL from CLI (Mid: 1)
1696746966: Sending PUBLISH to ditto (d0, q2, r0, m27, 'data', ... (5 bytes))
1696746966: Sending PUBCOMP to CLI (m1)
1696746966: Received DISCONNECT from CLI
1696746966: Client CLI disconnected.
...
1696747363: New connection from 127.0.0.1:44862 on port 1883.
1696747363: New client connected from 127.0.0.1:44862 as CLI (p5, c1, k60).
1696747363: No will message specified.
1696747363: Sending CONNACK to CLI (0, 0)
1696747363: Received PUBLISH from CLI (d0, q2, r0, m1, 'ditto', ... (170 bytes))
1696747363: Sending PUBREC to CLI (m1, rc0)
1696747363: Received PUBREL from CLI (Mid: 1)
1696747363: Sending PUBLISH to ditto (d0, q2, r0, m36, 'ditto', ... (170 bytes))
1696747363: Sending PUBCOMP to CLI (m1)
1696747363: Received DISCONNECT from CLI
1696747363: Client CLI disconnected.
```

Note that there is no acknowledgement from client "ditto".

Connection I used to test:

```json
{
    "connectionType": "mqtt-5",
    "connectionStatus": "open",
    "uri": "tcp://mosquitto:1883",
    "specificConfig": {
        "clientId": "ditto"
    },
    "sources": [
        {
            "addresses": [ "data" ],
            "qos": 2,
            "authorizationContext": ["nginx:ditto"],
            "payloadMapping": ["js-data"]
        },
        {
            "addresses": [ "ditto" ],
            "qos": 2,
            "authorizationContext": ["nginx:ditto"]
        }
    ],
    "targets": [],
    "mappingDefinitions": {
        "js-data": {
            "mappingEngine": "JavaScript",
            "options": {
                "incomingScript": " function mapToDittoProtocolMsg( headers, textPayload, bytePayload, contentType) { return Ditto.buildDittoProtocolMsg( 'test', 'thing-01', 'things', 'twin', 'commands', 'merge', '/features/mqtt/properties/data', { 'content-type': 'application/merge-patch+json' }, { headers, contentType, textPayload, }); }",
                "outgoingScript": ""
            }
        }
    }
}
```

If I leave only one source, everything works fine:

```text
1696747104: New connection from 127.0.0.1:40848 on port 1883.
1696747104: New client connected from 127.0.0.1:40848 as CLI (p5, c1, k60).
1696747104: No will message specified.
1696747104: Sending CONNACK to CLI (0, 0)
1696747104: Received PUBLISH from CLI (d0, q2, r0, m1, 'data', ... (4 bytes))
1696747104: Sending PUBREC to CLI (m1, rc0)
1696747104: Received PUBREL from CLI (Mid: 1)
1696747104: Sending PUBLISH to ditto (d0, q2, r0, m28, 'data', ... (4 bytes))
1696747104: Sending PUBCOMP to CLI (m1)
1696747104: Received DISCONNECT from CLI
1696747104: Client CLI disconnected.
1696747104: Received PUBREC from ditto (Mid: 28)
1696747104: Sending PUBREL to ditto (m28)
1696747104: Received PUBCOMP from ditto (Mid: 28, RC:0)
```

**Comment by thjaeckle** [comment]
@dimabarbul could you verify if this bug is now also resolved with #1794 being merged?
The latest `nightly` tag on Docker Hub should contain the fix.

**Comment by dimabarbul** [comment]
The bug is fixed in nightly and 3.4.2 images, but not in 3.4.1, which I guess is expected.

**Comment by thjaeckle** [comment]
Great to hear. Thank you again for the fix :+1:
